cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -315,6 +315,14 @@ func generateMountsModeRecover(mst *initramfsMountsState, recoverySystem string)
 		return err
 	}
 
+	// finally we need to modify the bootenv to mark the system as successful,
+	// this ensures that when you reboot from recover mode without doing
+	// anything else, you are auto-transitioned back to run mode
+	err = boot.MarkRecoverModeBootSuccessful(recoverySystem)
```

Comment:
(nitpick) - this could be written in a single line: `if err := boot.Mark....; err != nil {`